src/main/java/org/opentripplanner/middleware/controllers/api/OtpUserController.java

```diff
@@ -12,6 +15,15 @@ public OtpUserController(String apiPrefix) {
         super(apiPrefix, Persistence.otpUsers, "secure/user");
     }
 
+    @Override
+    OtpUser preCreateHook(OtpUser user, Request req) {
+        // Check API key and assign user to appropriate collection.
+        String apiKey = req.headers("x-api-key");
+        ApiUser apiUser = Persistence.apiUsers.getOneFiltered(Filters.eq("apiKeys.value", apiKey));
+        user.applicationId = apiUser.id;
```

Comment:
This needs a null check. ApiUser will be null if this `OtpUser` is signed up from MOD UI.

Reply:
Agree on the null check, but we should also change MOD UI to use an API key assigned to an ApiUser. If the ApiUser is null, we should halt.

Reply:
Creating a single ApiUser for MOD UI might make sense... however, even with that in place, the problem here is that for a fresh Auth0 id from a new signup from MOD UI, there will be no ApiUser for that Auth0 id.

Reply:
Fixed in 39bfee48488d070febab2255505565f5e875de37